src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceCreationService.kt

```diff
@@ -208,10 +203,20 @@ class LicenceCreationService(
     )
   }
 
-  private fun offenderHasLicenceInFlight(nomsId: String): Boolean {
-    val inFlight =
+  private fun verifyNoInFlightLicence(nomsId: String) {
+    val inflightLicences =
       licenceRepository.findAllByNomsIdAndStatusCodeIn(nomsId, listOf(IN_PROGRESS, SUBMITTED, APPROVED, REJECTED))
-    return inFlight.isNotEmpty()
+
+    if (inflightLicences.isNotEmpty()) {
+      val currentInflightLicence = when {
+        inflightLicences.size == 1 -> inflightLicences.first()
+        else -> inflightLicences.first { it.statusCode != APPROVED }
```

Comment:
if there's more than one inflight licence then there will only be two licences and one would be approved. 
We wouldn't want to choose the APPROVED one